src/components/LongTermAverageGraph/LongTermAverageGraph.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import {values, keys} from 'lodash';
+
+
+//this weird sequence avoids crashing npm when plotly is loaded.
+import Plotly from 'plotly.js';
+import createPlotlyComponent from 'react-plotly.js/factory';
+const Plot = createPlotlyComponent(Plotly);
+
+
+function LongTermAverageGraph({longTermData}) {
+
+    /* data is a temporary variable that represents the data, which are
+     key value pairs. The keys are the date of the measurement, the values
+     are the temperature at that date. The keys will be sorted, and then the
+     temperatures will be matched to the correct years. */
+    let data = longTermData.rXi1p1.data;   
+    const years = keys(data).sort();
+
+    /* function that returns an array of values, which are in the same order as
+     the sorted keys */
+    function matchValues(keys, values) {
```

Comment:
This function assumes that lodash's keys() and value() functions return things in matching order. That is, you're assuming that if you have an object like this: 
```javascript
const person = {
  firstName: "James",
  lastName: "Hiebert",
};
```

And you do 
```javascript
const k = keys(person);
const v = values(person);
```

you'll get 
```javascript 
=>  k
["firstName", "lastName"]

=> v 
["James", "Hiebert"]
```

but it's entirely possible to end up with:
```javascript
=>  k
["lastName", "firsttName"]

=> v 
["James", "Hiebert"]
```

If you look at the [lodash description of keys()](https://lodash.com/docs/4.17.15#keys) it says "Iteration order is not guarenteed." That is, they may come out in the "right" order in your browser, and in my browser (the graph looks fine to me) and in fact will probably work fine in most people's browsers,  but somewhere out there there's someone using an ancient version of Internet Explorer or something, and they're going to see `lastName` as `James` or see the order of the points on the graph scrambled because `keys()` and `values()` won't be in matching order for them.

Instead of figuring out what value goes with what date by using a list of `values` and assuming it's in the same order as your list of dates, you should be able to just use your ordered list of dates as keys to the original `data` object to get the values out of that object. As a bonus, this will be more efficient.